### PR TITLE
feat: a linear subtasking example in demo

### PR DESCRIPTION
**Pull request by ereali-aneo**

### CODE REVIEW

File: csharp/native/LinearSubTasking/Client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#See https://aka.ms/customizecontainer to learn how to customize your debug container and how Visual Studio uses this Dockerfile to build your images for faster debugging.
```
Comment (ngruelaneo) [inline csharp/native/LinearSubTasking/Client/Dockerfile:1]: a effacer

File: csharp/native/LinearSubTasking/Client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+FROM mcr.microsoft.com/dotnet/runtime:8.0 AS base
+#ARG TNAME="ArmoniK.Samples.LinearSubTasking.Client" TPATH="SubTasking/Client/"
```
Comment (ngruelaneo) [inline csharp/native/LinearSubTasking/Client/Dockerfile:5]: idem

File: csharp/native/LinearSubTasking/Client/Program.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                                         description: "Name of the partition to which submit tasks.",
+                                         getDefaultValue: () => "default");
+      var integer = new Option<int>("--integer",
+                                    description: "integer to % 2.",
+                                    getDefaultValue: () => 3);
+
+      // Describe the application and its purpose
+      var rootCommand =
+        new
+          RootCommand("Calculate the result of integer % 2 with subtask example for ArmoniK.\nIt sends a task to ArmoniK with integer in input and create a new subtask for each substraction of 2 while the result is superior to 1 ");
```
Comment (ngruelaneo) [inline csharp/native/LinearSubTasking/Client/Program.cs:184]:
```suggestion
          RootCommand("Calculate the result of integer % 2 with subtask example for ArmoniK.\nIt sends a task to ArmoniK with integer an as input and create a new subtask. At each subtask creation the number 2 is substract to the input integer until the result is greater than 1.");
```

**Review by ngruelaneo — APPROVED** [review]

File: csharp/native/LinearSubTasking/Client/ArmoniK.Samples.LinearSubTasking.Client.csproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Client/ArmoniK.Samples.LinearSubTasking.Client.csproj:6]:
```suggestion
```

Do not use implicit usings ! (in both projects^^)

File: csharp/native/LinearSubTasking/Worker/Program.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using ArmoniK.Api.Worker.Utils;
+using ArmoniK.Samples.LinearSubTasking.Worker;
+
+// Create the application and register thesubmit task example worker implementation
```
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Worker/Program.cs:28]:
```suggestion
// Create the application and register the submit task example worker implementation
```

File: csharp/native/LinearSubTasking/Worker/appsettings.Development.json
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Worker/appsettings.Development.json:1]: No need for this file. Can you remove it from history ?

File: csharp/native/LinearSubTasking/Client/Program.cs [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                                                          Results =
+                                                          {
+                                                            new CreateResultsMetaDataRequest.Types.ResultCreate
+                                                            {
+                                                              Name = "Result",
+                                                            },
+                                                          },
+                                                        })
+                                 .Results.Single()
+                                 .ResultId;
```
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Client/Program.cs:114]: Can you print this resultId ? (mainly for easier debugging when there is an issue during task execution)

File: csharp/native/LinearSubTasking/Client/Program.cs [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+                                         description: "Name of the partition to which submit tasks.",
+                                         getDefaultValue: () => "default");
+      var integer = new Option<int>("--integer",
+                                    description: "integer to % 2.",
+                                    getDefaultValue: () => 3);
+
+      // Describe the application and its purpose
+      var rootCommand =
+        new
+          RootCommand("Calculate the result of integer % 2 with subtask example for ArmoniK.\nIt sends a task to ArmoniK with integer an as input and create a new subtask. At each subtask creation the number 2 is substract to the input integer as long as the result is greater than 1.");
```
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Client/Program.cs:184]:
```suggestion
          RootCommand("Calculate the result of integer % 2 with subtask example for ArmoniK.\nIt sends a task to ArmoniK with integer an as input and create a new subtask. At each subtask creation the number 2 is substracted to the input integer as long as the result is greater than 1.");
```

File: csharp/native/LinearSubTasking/Worker/LinearSubTasking.cs [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                              input,
+                              resultId);
+        }
+        else if (input < 0)
+        {
+          input = input + 2;
+          await SubmitWorkers(taskHandler,
+                              input,
+                              resultId);
+        }
```
Comment (aneojgurhem) [inline csharp/native/LinearSubTasking/Worker/LinearSubTasking.cs:65]: Can and should be factorized a little bit

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

**Pull request closed without merging**